build.gradle

```diff
@@ -208,92 +207,16 @@ task setupEclipse {
     dependsOn createEclipseLaunches
 }
 
-/*
-idea {
-    module {
-        // Removes the source directories added by the appengine plugin
-        testSourceDirs -= sourceSets.functionalTest.allSource.srcDirs
-    }
-}
-
-// removes unused sourceSets from the appengine plugin
-// the appengine plugin also adds source directories to the IDEA plugin, which are removed from the idea block above
-sourceSets.remove(sourceSets.functionalTest)
-sourceSets.remove(sourceSets.endpointsSrc)
-
 String intellijSetupGroup = 'IntelliJ IDEA Setup'
 
 // IntelliJ Project Setup
-import groovy.xml.StreamingMarkupBuilder
-
-task addAppEngineSdkHomePath(dependsOn: appengineDownloadSdk) {
-    group intellijSetupGroup
-    doLast {
-        String moduleXmlFilePath = '.idea/modules/teammates.iml'
-        Node moduleXml = new XmlParser().parse(moduleXmlFilePath)
-        Node facetComponent = (Node) (moduleXml.component.find { it.@name == 'FacetManager' })
-        Node googleAppEngineFacet = (Node) (facetComponent.find { it.@type == 'google-app-engine' })
-        Node configuration = googleAppEngineFacet.configuration[0]
-
-        NodeList sdkHomePathList = configuration.'sdk-home-path'
-        if (sdkHomePathList.size() > 1) {
-            throw new GradleException('More than one sdk-home-path found in Google App Engine Facet')
-        } else if (sdkHomePathList.size() == 1) {
-            configuration.'sdk-home-path'[0].value = System.properties['appengine.sdk.root']
-        } else {
-            List configurationChildren = configuration.children()
-            configurationChildren.add(0, new Node(null, 'sdk-home-path', null, System.properties['appengine.sdk.root']))
-        }
-
-        new File(moduleXmlFilePath).withWriter('UTF-8') { out ->
-            out << new StreamingMarkupBuilder().bind { mkp.xmlDeclaration() }
-            def xmlNodePrinter = new XmlNodePrinter(new PrintWriter(out))
-            xmlNodePrinter.preserveWhitespace = true
-            xmlNodePrinter.print(moduleXml)
-        }
-    }
-}
-
-task deleteIdeaFolder(type: Delete) {
-    delete '.idea'
-}
-
-task copyIntelliJProjectSettings(type: Copy) {
-    group intellijSetupGroup
-    if (file('.idea').exists()) {
-        if (!file('.idea').isDirectory()) {
-            throw new GradleException('Found file with name .idea where it should be a folder.' +
-                    'Please rename or delete it otherwise setup cannot proceed.')
-        }
-
-        if (project.hasProperty('delete_previous_idea')) {
-            println 'Your previous project will be overwritten.'
-            dependsOn deleteIdeaFolder
-        } else {
-            doLast {
-                throw new GradleException('A project folder is found, you can delete it manually or ' +
-                        'call the setup again with the property -Pdelete_previous_idea')
-            }
-        }
-    }
-    from '.templates/.idea'
-    into '.idea'
-}
 
 task setupIntellijRunConfigs(type: Copy) {
```

Comment:
You may want to add `setupIntellij` task that simply calls this, much like `setupEclipse` up there.